File: php/features.php
<?php
declare(strict_types=1);

// WaybackMachine SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class WaybackMachineFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new WaybackMachineBaseFeature();
            case "test":
                return new WaybackMachineTestFeature();
            default:
                return new WaybackMachineBaseFeature();
        }
    }
}

File: py/features.py
# WaybackMachine SDK feature factory

from feature.base_feature import WaybackMachineBaseFeature
from feature.test_feature import WaybackMachineTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WaybackMachineBaseFeature(),
        "test": lambda: WaybackMachineTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
